airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -812,6 +812,11 @@ definitions:
         description: Set to True if the target API does not accept queries where the start time equal the end time.
         type: boolean
         default: False
+      global_parent_cursor:
```

Comment:
I think we should try to speak the user's language here. 

How about we call this
`use_parent_cursor`

`title: Whether to use the parent stream cursor`

`description: If the parent stream cursor is kept in sync with this stream's, using the parent stream's cursor allows for more efficient storage of the connection state.`

@lmossman thoughts on how make this most useful for users?

Reply:
I think I'd modify this slightly:
```
use_parent_cursor:
  title: Use Parent Cursor
  description: Make syncs more efficient by using the parent stream's cursor for the connection state.\n\n**WARNING:** This only works if both the parent stream and this child stream are incremental and if the parent stream cursor is kept in sync with this child stream's cursor!
```
If that sounds correct to you.

Though thinking about this more, I almost feel like we should just have a single boolean field in the builder like `Parent cursor is updated when child cursor is updated?`. And if the user enables this, then we set both `incremental_dependency` and `use_parent_cursor` to `true`, since I think they would always want both enabled in that case, and that way the user doesn't really need to know about the internals.

What do you think?

Reply:
Alternatively, what if we just combined this `global_parent_cursor` functionality into the existing `incremental_dependency` field? Is there ever a use case to have one enabled and not the other?

Reply:
@girarda, @lmossman That's not the correct explanation for this parameter. This is what incremental_dependency does. The global cursor is responsible for storing the state as one value instead of per partition.

These parameters can be used separately:

- `incremental_dependency: true + global_cursor: true`: Incremental streams with dependency.
- `incremental_dependency: true + global_cursor: false`: Per partition but saves state for efficiency, need a lookback.
- `incremental_dependency: false + global_cursor: true`: Not a good idea, as we will always read the parent in full refresh.
- `incremental_dependency: false + global_cursor: false`: Per partition with full refresh parent.

Reply:
What is the use case for this option?
> `incremental_dependency: true + global_cursor: false`: Per partition but saves state for efficiency, need a lookback.

I thought that the requirement that updates to the child record resulted in updates to the parent record cursor was just for the `incremental_dependency` value, and if so it seems like `global_cursor: true` would always be desired in that situation, right?

Reply:
The use case involves a stream without dependency between cursors. Adding `incremental_dependency` may lead to losing records but will help limit parent records, significantly speeding up the sync compared to using just PerPartition without parent state.

Reply:
But in that use case, why would you ever want to have `global_cursor: false`? It seems like you would always want to have `global_cursor: true` if you are setting `incremental_dependency: true`, so that the state is saved more efficiently, right? I feel like I'm missing something here

Reply:
For example, we can use `incremental_dependency` without a global cursor to avoid using the lookback window. This approach may be more efficient if the number of partitions is small.

Reply:
I'm confused, you said above:
> `incremental_dependency: true + global_cursor: false`: Per partition but saves state for efficiency, need a lookback.

But just now you said 
> we can use `incremental_dependency` without a global cursor to avoid using the lookback window

Which seems contradictory. 

It doesn't seem like this would be a straightforward decision for users to make, since I am also struggling to understand the use cases. Is there no way we can simplify this more?

Reply:
Sorry for the confusion, I mixed up different cases. [I've added more documentation for a better explanation.](https://github.com/airbytehq/airbyte/blob/81bbd3ef8fd7ee4ecb80132f7c42da4123244f95/docs/connector-development/config-based/understanding-the-yaml-file/partition-router.md) Here's a short recap:

- **Global Cursor:** This should be used if there are many partitions, as we cannot store all partitions, so there's no reason to save only part of them.
- **Incremental Dependency:** This should be used if there is a dependency between parent and child cursors. Without the dependency, we can miss new child records that were added to old parent records.

However, previously and possibly in the future, we used the incremental dependency approach for streams without dependency. Although this can lead to missing records, it provides a way to read these streams incrementally, significantly increasing performance.